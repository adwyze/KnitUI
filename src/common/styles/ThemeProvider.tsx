import React, { ReactElement } from "react"
import { ThemeProvider as Theme } from "styled-components"
import { generateTheme, palette as defaultPalette } from "./"

interface ThemeProviderProps {
  children: ReactElement
  theme?: object
}

const ThemeProvider = ({
  children,
  theme = generateTheme(defaultPalette),
}: ThemeProviderProps): ReactElement => <Theme theme={theme}>{children}</Theme>

export default ThemeProvider
