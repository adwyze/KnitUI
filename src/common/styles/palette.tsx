/* Primary Palette
---------------
The primary palette comprises bright and bold range of hues derived from Lapis lazuli. These should be used selectively to highlight the important actions and guide the eye.
*/

export const Blue100 = {
  hsl: [216, 100, 20],
  hex: "#002966",
}

export const Azure80 = {
  hsl: [205, 85, 65],
  hex: "#5AB2F2",
}

export const LightBlue40 = {
  hsl: [203, 100, 95],
  hex: "#E5F5FF",
}

export const Neutral0 = {
  hsl: [0, 0, 100],
  hex: "#FFFFFF",
}

/* Secondary Palette
--------------------
The secondary pallete comprises composed hues that compliment the primaries well. These provide a warmer tone to the product, and are deeply associated with the state and type of action, for instance, Crimson - danger/failure, Dark Lime - success, Pure Yellow - warning alerts, Moderate magenta - unsaved state.
*/

export const Crimson80 = {
  hsl: [0, 100, 30],
  hex: "#990000",
}

export const Green80 = {
  hsl: [118, 100, 20],
  hex: "#036600",
}

export const Yellow80 = {
  hsl: [47, 100, 50],
  hex: "#FFC700",
}

export const Magenta80 = {
  hsl: [288, 20, 35],
  hex: "#64476B",
}

/* Tertiary Palette
---------------
The tertiary palette comprises the washed-out hues that are used for highlights. It provides a softer experience and help in easily differentiating between bits.
*/

export const Beige10 = {
  hsl: [42, 38, 95],
  hex: "#F7F4ED",
}

export const Rose10 = {
  hsl: [0, 100, 95],
  hex: "#FFE3E3",
}

export const Fern10 = {
  hsl: [148, 33, 90],
  hex: "#DDEEE5",
}

export const Sun10 = {
  hsl: [40, 92, 90],
  hex: "#FDEDCE",
}

export const Sapphire10 = {
  hsl: [217, 80, 90],
  hex: "#D1E1FA",
}

export const Violet10 = {
  hsl: [274, 49, 90],
  hex: "#E7D9F2",
}

export const Coral10 = {
  hsl: [22, 80, 90],
  hex: "#FAE0D1",
}

export const Wood10 = {
  hsl: [33, 35, 85],
  hex: "#E6DACB",
}

/* Neutral Palette
---------------
The neutrals bring balance to the color palette, and used for actions that either donâ€™t need direct attention or reduce focus from those already taking more than needed. Darker neutrals are used for shadows, borders, etc.
*/

export const Neutral05 = {
  hsl: [0, 0, 97],
  hex: "#F7F7F7",
}

export const Neutral10 = {
  hsl: [0, 0, 95],
  hex: "#F2F2F2",
}

export const Neutral20 = {
  hsl: [0, 0, 90],
  hex: "#E5E5E5",
}

export const Neutral30 = {
  hsl: [0, 0, 85],
  hex: "#D9D9D9",
}

export const Neutral40 = {
  hsl: [0, 0, 80],
  hex: "#CCCCCC",
}

export const Neutral45 = {
  hsl: [0, 0, 65],
  hex: "#A6A6A6",
}

export const Neutral50 = {
  hsl: [0, 0, 50],
  hex: "#808080",
}

export const Neutral60 = {
  hsl: [0, 0, 40],
  hex: "#666666",
}

export const Neutral70 = {
  hsl: [0, 0, 30],
  hex: "#4C4C4C",
}

export const Neutral80 = {
  hsl: [0, 0, 20],
  hex: "#333333",
}

export const Neutral90 = {
  hsl: [0, 0, 10],
  hex: "#1A1A1A",
}

export const Azure100 = {
  hsl: [205, 85, 40],
  hex: "#0F74BD",
}
