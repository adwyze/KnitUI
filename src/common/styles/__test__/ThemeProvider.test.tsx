import { render } from "@testing-library/react"
import React from "react"
import { Button } from "../../.."
import * as palette from "../palette"
import ThemeProvider from "../ThemeProvider"
import generateTheme from "../variables"

describe("Passing Custom Theme", () => {
  // Changing one of the palette color
  const newPalette = {
    ...palette,
    Blue100: {
      hsl: [205, 85, 45],
      hex: "#1182D4",
    },
  }

  const knituiTheme = generateTheme(newPalette)

  it("Changing palette color Blue100 to #1182D4", () => {
    const { asFragment } = render(
      <ThemeProvider theme={knituiTheme}>
        <Button kind="primary" size="small" label="Changed Color of Neutral" />
      </ThemeProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
