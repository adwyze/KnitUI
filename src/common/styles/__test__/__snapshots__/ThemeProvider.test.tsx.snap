// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passing Custom Theme Changing palette color Blue100 to #1182D4 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.2rem;
  line-height: 1.8rem;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
  color: #ffffff;
  background-color: #1183d4;
  padding-top: 0.1rem;
  padding-bottom: 0.1rem;
  border-radius: 0.4rem;
  border-style: none;
  box-sizing: border-box;
  border: 1px solid #00000000;
}

.c0 span {
  margin-right: 0rem;
}

.c0 span svg path {
  fill: #ffffff;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #ffffff;
  background-color: #2a9dee;
  cursor: pointer;
}

.c0:hover svg path,
.c0:active svg path,
.c0:focus svg path {
  fill: #ffffff;
}

.c0:active,
.c0:focus {
  border: 1px solid #5ab2f2 !important;
  box-shadow: 0px 0px 2px #0066ff;
  outline: none;
}

.c0:hover {
  border: 1px solid #00000000;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    class="c0"
  >
    Changed Color of Neutral
  </button>
</DocumentFragment>
`;
